src/service/referral.service.ts

```diff
@@ -10,6 +10,7 @@ import OpeningService from "../service/opening.service";
 import { ReferralStatus } from "../utils/status.enum";
 import { PermissionLevel } from "../utils/permission.level.enum";
 import { Like } from "typeorm";
+import OpeningRepository from "../repository/opening.repository";
```

Comment:
It is not a good practice to use the repository layer of another entity. Instead use the service layer of Opening entity.